### PR TITLE
#2705 Add DisplayCurrency component and tests.

### LINKED ISSUE #2705: Fix negative numbers in invoices
Negative numbers should be displayed like ($10), instead of as $-10.

### PR DESCRIPTION
**Pull request by rjstires**
Closes #2705 

Created a component for use across project for displaying currencies. Can be easily extended to support additional currencies if that comes up later.

With the collective 👍  I can apply across the project.

### CODE REVIEW

File: src/components/DisplayCurrency.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * recommendation on where to store it.
+ */
+const ifElse = (pred, ifFn, elseFn) => (value) => pred(value)
+  ? ifFn(value)
+  : elseFn(value);
+
+const normalizeNumber = (v) => Math.abs(v).toFixed(2);
+
+const convertToCurrency = ifElse(
+  (v) => v < 0,
```
Comment (displague) [inline src/components/DisplayCurrency.js]: Is there a general benefit to an ifElse over just doing:
```
convertToCurrency = (v) => v<0?`one way`:`or another`;
```

Comment (rjstires) [reply src/components/DisplayCurrency.js]: Honestly, no. I'd created this when I was using function composition, but then pulled back. I can clean this up a bit.

**Comment by rjstires** [comment]
Updated per feedback.

**Comment by asauber** [comment]
If we ever do currency calculations on the frontend (do we do any now?), we should use a Currency type that separates whole currency and fractional currency into two integers.

Repeated calculations with floating-point currency results in inaccuracies because of floating point error.

e.g. 

```
1.03 - 0.42
0.6100000000000001
```

In the case that we do perform calculations, I would expect this component to accept a `dollars` prop and a `cents` prop. Until then, this LGTM.

**Comment by asauber** [comment]
Values for the "Amount" column in the table are still formatted incorrectly.

https://github.com/rjstires/manager/blob/680819062603b6b8014573c4e2034ecefd309ebd/src/billing/layouts/InvoicePage.js#L91

screen shot 2017-11-20 at 11 51 45 am

**Comment by rjstires** [comment]
Once @displague approves this method, Ill apply it across the project.

**Comment by na3d** [comment]
Go ahead and implement the component everywhere.  It is definitely a good solution, and will make our lives easier going forward.

**Comment by rjstires** [comment]
@displague @na3d Ready for any final comments/review.

**Comment by na3d** [comment]
Getting [object Object] instead of the amount for invoice items.

**Pull request merged**